ptrace.c

```diff
@@ -96,19 +96,22 @@ static struct {
 
 bool sm_attach(pid_t target)
 {
-    int status;
-
-    /* attach to the target application, which should cause a SIGSTOP */
-    if (ptrace(PTRACE_ATTACH, target, NULL, NULL) == -1L) {
-        show_error("failed to attach to %d, %s\n", target, strerror(errno));
-        return false;
-    }
+    if (!sm_globals.options.no_ptrace)
```

Comment:
You don't issue sigstop at all here in no ptrace mode.
I've just tested it, and it works, but I thought a process had to be stopped for you to read/write to its procmem.

Reply:
Since linux kernel >= 3.2.2 it is using the memory mapping directly with copy_from_user/copy_to_user functions, which removes the need of the process being stopped I assume, because the kernel is probably stopping the process directly when it tries to read when it uses the mmap semaphores

References: 
https://elixir.bootlin.com/linux/v3.2.2/source/fs/proc/base.c#L829
https://elixir.bootlin.com/linux/v3.2.2/source/fs/proc/base.c#L853
https://elixir.bootlin.com/linux/v3.2.2/source/mm/memory.c#L3828
https://elixir.bootlin.com/linux/v3.2.2/source/mm/memory.c#L3766

Then it translates the vma into page where you can convert after into kernel address, copy_from_user/copy_to_user, etc..

On linux kernel < 3.2.2 there is a function called check_mem_permission, which decides if the task has permissions to read/write to the targeted task memory
https://elixir.bootlin.com/linux/v3.2.1/source/fs/proc/base.c#L216

Which needs the targeted task to be stopped.